src/main/java/com/o19s/es/explore/ExplorerQuery.java

```diff
@@ -104,11 +105,11 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
 
             for (Term term : terms) {
                 TermStates ctx = TermStates.build(searcher.getTopReaderContext(), term, scoreMode.needsScores());
-
-                if(ctx != null){
-                    df_stats.add(ctx.docFreq());
-                    idf_stats.add(sim.idf(ctx.docFreq(), searcher.collectionStatistics(term.field()).docCount()));
-                    ttf_stats.add(ctx.totalTermFreq());
+                if(ctx != null && ctx.docFreq() > 0){
+                    TermStatistics tStats = searcher.termStatistics(term, ctx.docFreq(), ctx.totalTermFreq());
```

Comment:
I dug into TermStatistics yesterday and it appears to just be a wrapper around the values you pass in for docFreq and ttf.  Can you provide an example where we would get different values from using `TermStatistics::docFreq` over using `TermStates::docFreq`?

Reply:
I guess the problem is` TermStatistics.termStatistics` can be [overridden](https://lucene.apache.org/core/8_9_0/core/org/apache/lucene/search/IndexSearcher.html#termStatistics-org.apache.lucene.index.Term-int-long-). Lucene says: "This can be overridden for example, to return a term's statistics across a distributed collection." 
and that behavior doesn't happen to `TermStates`. I will provide an example in a bit.

Reply:
Okay I see one example at https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/dfs/DfsPhase.java#L37 which makes sense.  

If it's not a pain it would be interesting to get tests built around these cases and to possibly support both shard/index level stats via a param.

Reply:
Hey Dan,
I provided an reproducible example [here](https://github.com/o19s/elasticsearch-learning-to-rank/issues/381).

Reply:
I also see an "overridden" example here: https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java#L279

Reply:
Makes sense, I've updated the TermStatSupplier as well.